java/client/src/main/java/glide/api/commands/SortedSetBaseCommands.java

```diff
@@ -728,4 +729,91 @@ CompletableFuture<Long> zrangestore(
      * }</pre>
      */
     CompletableFuture<Long> zlexcount(String key, LexRange minLex, LexRange maxLex);
+
+    /**
+     * Returns the union of members from sorted sets specified by the given <code>keys</code>.<br>
+     * To get the elements with their scores, see {@link #zunionWithScores}.
+     *
+     * @apiNote When in cluster mode, all <code>keys</code> must map to the same <code>hash slot
+     *     </code>.
+     * @see <a href="https://redis.io/commands/zunion/">redis.io</a> for more details.
+     * @param keys The keys of sorted sets.
+     * @param options Weight and Aggregate options.
+     * @return The resulting sorted set from the union.
+     * @example
+     *     <pre>{@code
+     * WeightAggregateOptions options =
+     *     WeightAggregateOptions.builder()
+     *             .aggregate(Aggregate.MAX)
+     *             .weight(1.0)
+     *             .weight(2.0)
+     *             .build();
+     * String[] payload = client.zunion(new String[] {"mySortedSet1", "mySortedSet2"}, options).get()
+     * assert payload.equals(new String[] {"elem1", "elem2", "elem3"});
+     * }</pre>
+     */
+    CompletableFuture<String[]> zunion(String[] keys, WeightAggregateOptions options);
```

Comment:
I feel like this implementation can raise some problems.
from what I know, when specifying weights, each key must get its own weight, what means that len(keys) == len(weights) and we shouldnt allow clients to pass different length of each.

I think we should split the function into 2, one that only gets keys and another one that gets a tuple (pair in java?)  of keys and weights.

Reply:
I see theres a test for it, so this is not good, we can overcome it

Reply:
we can also think about a special struct for it that wont require us to split the function into 2 cases

Reply:
Should we update `zadd` and `set`, etc as well to combine all command arguments into one structure?
https://github.com/aws/glide-for-redis/blob/3b29627e2fae645df75cf6545cbdef561a456307/java/client/src/main/java/glide/api/commands/SortedSetBaseCommands.java#L52-L53
https://github.com/aws/glide-for-redis/blob/3b29627e2fae645df75cf6545cbdef561a456307/java/client/src/main/java/glide/api/commands/StringBaseCommands.java#L71

Reply:
I think zadd is different because its good that we wont really allow executing zadd with the same key twice, so map is good.

about set I am not sure, we should keep the existing API that matches the other wrappers

Reply:
You could add an explicit check in the function for it to return early but don't see the exact point in that if redis is going to do the check for you.

Reply:
what about creating an abstract class with 2 subclasses? one that holds just the keys, and another one to hold keys and weights?

or maybe a class that holds the keys and a subclass that also holds weights

Reply:
I believe adding these will make this API more complex than it needs to be. Options have been the standard so far in Java (except zrange) and deferring from it could be confusing. This would also affect python and node because the same functionality would also need to be implemented there (which have been using optional values so far). 

Abstract classes could be solution but it doesn't feel like the same API we have had so far. This is something being explicitly checked on Redis so don't know if this should be something we should be reinforcing on the client side.

Reply:
in python we'll probably add typing of Union[str, Tuple[str, float]], but I dont know what to decide about java

@barshaul what do you think?

Reply:
One of our API design decisions is to utilize the type system for user input validation in non-generic methods, and we should maintain this approach consistently across all functions. Let's implement two functions: one that accepts keys as a list of strings, and another that takes a list of key-weight pairs. I know it creates a large functions matrix for this command in Java, but most of the commands don't require such extensive handling

Reply:
This won't just be for this command there are a couple of commands that use Weights and Agregates and this function alone will need 8 signatures. Are you sure about this approach?

Reply:
Yes. this is one of the guidelines for GLIDE's API

Reply:
Sorry for the short message - wasn't near my PC.
We chose to use the type system to catch errors at compile time rather than when the code runs, aiming to guide users towards correct client usage. By exposing well-defined methods, we reduce the need for extensive verification—the type system handles it for us. In other languages, like Python and Node, this verification can be done in a single function, as Shoham mentioned, and since we aim for a consistent API across all languages - if we allow mistakes in one wrapper—like passing arrays with different lengths—we'll need to do the same in other wrappers too. 
Given that most commands don't require as many signatures, and it's easily achievable in other languages, and the previously discussed pros/cons for of letting the server return with an error / raising errors in runtime from the client side - I don't think the multiple signatures in some commands is a strong enough reason to deviate from our current design guideline